src/engine/bank05.asm

```diff
@@ -54,31 +54,1059 @@ PointerTable_14000: ; 14000 (05:4000)
+.add_poison
+	ld a, c
+	add e
+	ld e, a
+	ld a, $00
+	adc d
+	ld d, a
+.not_poisoned
+	call SwapTurn
+
+.done
+	pop hl
+	ld [hl], e
+	ld a, d
+	or a
+	ret z
+	; cap damage
+	ld a, $ff
+	ld [hl], a
+	ret
+; 0x1450b
+
+; stores in wDamage, wAIMinDamage and wAIMaxDamage the calculated damage
+; done to the Pokémon at hTempPlayAreaLocation_ff9d
+; by the defending Pokémon, using the move index at a
+; input:
+; 	a = move index
+;	[hTempPlayAreaLocation_ff9d] = location of card to calculate
+;								 damage as the receiver
+CalculateMoveDamage_FromDefendingPokemon: ; 1450b (5:450b)
+	call SwapTurn
+	ld [wSelectedMoveIndex], a
+	ld e, a
+	ld a, DUELVARS_ARENA_CARD
+	call GetTurnDuelistVariable
+	ld d, a
+	call CopyMoveDataAndDamage_FromDeckIndex
+	call SwapTurn
+	ld a, [wLoadedMoveCategory]
+	cp POKEMON_POWER
+	jr nz, .is_move
+
+; is a Pokémon Power
+; set wDamage, wAIMinDamage and wAIMaxDamage to zero
+	ld hl, wDamage
+	xor a
+	ld [hli], a
+	ld [hl], a
+	ld [wAIMinDamage], a
+	ld [wAIMaxDamage], a
+	ld e, a
+	ld d, a
+	ret
+
+.is_move
+; set wAIMinDamage and wAIMaxDamage to damage of move
+	ld a, [wDamage]
+	ld [wAIMinDamage], a
+	ld [wAIMaxDamage], a
+	call SwapTurn
+	ldh a, [hTempPlayAreaLocation_ff9d]
+	push af
+	xor a
+	ldh [hTempPlayAreaLocation_ff9d], a
+	ld a, EFFECTCMDTYPE_AI
+	call TryExecuteEffectCommandFunction
+	pop af
+	ldh [hTempPlayAreaLocation_ff9d], a
+	call SwapTurn
+	ld a, [wAIMinDamage]
+	ld hl, wAIMaxDamage
+	or [hl]
+	jr nz, .calculation
+	ld a, [wDamage]
+	ld [wAIMinDamage], a
+	ld [wAIMaxDamage], a
+
+.calculation
+; if temp. location is active, damage calculation can be done directly...
+	ldh a, [hTempPlayAreaLocation_ff9d]
+	or a
+	jr z, CalculateDamage_FromDefendingPokemon
+
+; ...otherwise substatuses need to be temporarily reset to account
+; for the switching, to obtain the right damage calculation...
+	ld a, DUELVARS_ARENA_CARD_SUBSTATUS1
+	call GetTurnDuelistVariable
+	push af
+	push hl
+	ld [hl], $00
+	; reset substatus2
+	ld l, DUELVARS_ARENA_CARD_SUBSTATUS2
+	ld a, [hl]
+	push af
+	push hl
+	ld [hl], $00
+	; reset changed resistance
+	ld l, DUELVARS_ARENA_CARD_CHANGED_RESISTANCE
+	ld a, [hl]
+	push af
+	push hl
+	ld [hl], $00
+	call CalculateDamage_FromDefendingPokemon
+; ...and subsequently recovered to continue the battle normally
+	pop hl
+	pop af
+	ld [hl], a
+	pop hl
+	pop af
+	ld [hl], a
+	pop hl
+	pop af
+	ld [hl], a
+	ret
+
+; similar to CalculateDamage_VersusDefendingPokemon but reversed,
+; calculating damage of the defending Pokémon versus
+; the card located in hTempPlayAreaLocation_ff9d
+; taking into account weakness/resistance/pluspowers/defenders/etc
+; and poison damage for two turns
+; and outputs the result capped at a max of $ff
+; input:
+; 	[wAIMinDamage] = base damage
+; 	[wAIMaxDamage] = base damage
+; 	[wDamage]      = base damage
+;	[hTempPlayAreaLocation_ff9d] = location of card to calculate
+;								 damage as the receiver
+CalculateDamage_FromDefendingPokemon: ; 1458c (5:458c)
```

Comment:
Counterpart to ``CalculateMoveDamage_VersusDefendingCard``. If you make the adjustments I suggested for that one, I will do the same in this one (would be ``CalculateMoveOrPokemonPowerDamage_VersusDefendingCard`` or ``EstimateDamage_VersusDefendingCard``, for example.